### PR TITLE
240 split stocktake adjustment

### LINKED ISSUE #240: Stock take: Reducing stock creates ledger problem
### Steps to reproduce

- Choose the item that does not have ledger problem
- Make a Stock take -> Add stock adding line -> Click `Confirm Finalised`
- Wait to sync, Once synced, Stock should be introduced for that item
- Do a next stock take, This time Reduce the stock
- Click `Confirm Finalised` -> Wait to sync
- Once synced to central server, Check that item -> stock
- Do ledger check, Ledger problem will be seen

**Branch:** _Main_

Video:
https://www.dropbox.com/s/cjvxj5qdlcqews8/Ledger%20problem.mp4?dl=0

The problem is: The reduced stock is also treated as a stock In type meaning `transaction type = si` over `type = sc` seeing that, Maybe it was not on the Roster.... ?

### PR DESCRIPTION
**Pull request by andreievg**
closes #240 

* Replaced [stocktake].inventory_adjustment_id and INVENTORY_ADJUSTMENT [invoice_type] with inventory `addition`, also added [stocktake]inventory_reduction_id and INVENTORY_REDUCTION [invoice_type]
* Changed stocktake service to add two inventory adjustments
* Updated sync layer to pull and push two types of inventory adjustments, had to add legacy `sc` type.

`note` I could have gone further with migrations and split existing inventory adjustments, but thought its not worth it, as long as upgrade doesn't break existing dev data files and doesn't stop sync or re-sync from existing sites.

TODO (make issue) to makes sure integration tests compile and pass -> https://github.com/openmsupply/open-msupply/issues/1045

### CODE REVIEW

File: server/repository/src/db_diesel/invoice_row.rs [hunk 1]
```diff
@@ -49,7 +49,11 @@ joinable!(invoice -> user_account (user_id));
 pub enum InvoiceRowType {
     OutboundShipment,
     InboundShipment,
-    InventoryAdjustment,
+    // Initially we had single inventory adjustment InvoiceRowType, this was changed to two seperate types
+    // central server may have old inventory adjustement type, thus map it to inventory additions
+    #[serde(alias = "INVENTORY_ADJUSTMENT")]
```
Comment (andreievg) [inline server/repository/src/db_diesel/invoice_row.rs:54]: Doing this in sync translation layer would require a lot more work (another invoice row type with manual mapping), not sure happy that it's in repository layer though, I wish we could auto derive things for  other crates within workspace

File: server/repository/src/migrations/v1_01_01/mod.rs [hunk 2]
```diff
@@ -32,6 +33,42 @@ impl Migration for V1_01_01 {
     }
 }
 
+#[cfg(test)]
+async fn setup_data_migration(name: &str) -> StorageConnection {
```
Comment (andreievg) [inline server/repository/src/migrations/v1_01_01/mod.rs:37]: this was used in `remove check from sqlite` work, but also would be needed in tests for splitting inventory adjustment migration, so relocated here

File: server/service/src/sync/test/test_data/invoice.rs [hunk 3]
```diff
@@ -467,18 +468,340 @@ fn transact_om_fields_push_record() -> TestSyncPushRecord {
...
             om_status: Some(InvoiceRowStatus::Shipped),
-            om_type: Some(InvoiceRowType::InventoryAdjustment),
+            om_type: Some(InvoiceRowType::InventoryAddition),
             om_colour: Some("SomeColour".to_string()),
             tax: Some(0.0),
         }),
     }
 }
 
+const INVENTORY_ADDITION: (&'static str, &'static str) = (
```
Comment (andreievg) [inline server/service/src/sync/test/test_data/invoice.rs:478]: Data is from mSupply inventory adjustment records

**Comment by mark-prins** [comment]
I created a stocktake

image

but the quantity does not appear in desktop..?

Screenshot 2023-01-17 at 10 48 52 AM

**Comment by mark-prins** [comment]
> but the quantity does not appear in desktop..?

User error - was confused because stock take date wasn't populated. Have tried with a simpler datafile and the functionality is good, and have retested using the original and that's fine too

**Review by mark-prins — APPROVED** [review]
LGTM! 🎉

**Comment by andreievg** [comment]
Thanks @mark-prins, I did another quick test too with demo data file, seems to be fine (and both inventory adjustments show in ledger)

**Pull request merged**